libraries/TGRSIAnalysis/TAngularCorrelation/TAngularCorrelation.cxx

```diff
@@ -697,7 +1056,146 @@ std::map<Int_t,std::map<Int_t,Int_t>> TAngularCorrelation::GenerateIndexMap(std:
 
    return indexmap;
 }
+////////////////////////////////////////
+/// Assign Groups for Angular Indexes
+///
+/// \param[in] group vector of assigned groups for angular indexes
+/// \param[in] anglemap
+
+std::vector<Int_t> TAngularCorrelation::AssignGroups(std::vector<Int_t> &group, std::vector<Double_t> &anglemap)
+{
+   std::vector<Int_t> groups; // vector to return
+
+   // get number of group entries
+  Int_t size = group.size();
+
+// basic consistency check
+  if (size!=(Int_t) anglemap.size()) {     
+ } 
+   
+   // initialize group vector
+   for (Int_t i=0;i<size;i++) {
+      groups.push_back(group[i]);
+   }
+
+   return groups;
+}
+////////////////////////////////////////
+/// Assign average angles for groups
+///
+/// \param[in] groupangle vector as assigned by user
+/// \param[in] groupweights vector for consistancy check
+
+std::vector<Double_t> TAngularCorrelation::AssignGroupAngles(std::vector<Double_t> &groupangles, std::vector<Int_t> &groupweights)
+{
+   std::vector<Double_t> groupangle; // vector to return
+
+   // get number of group entries
+   Int_t size = groupangles.size();
+
+   // basic consistency check
+   if(size!=(Int_t) groupweights.size()) {
+      printf("Not all groups have been assigned an angle.\n");   
+   } 
+   
+   // initialize group vector
+   for(Int_t i=0;i<size;i++) {
+       groupangle.push_back(groupangles[i]);
+   }
+
+   return groupangle;
+}
```

Comment:
groupangle also needs to be assigned to a class member so we can keep it around.